From 8a32c9514c4db6e242e338ca11b854c7760f4fa2 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Mon, 7 Dec 2015 10:58:26 +0800
Subject: [PATCH 08/13] ov8858: modify ov8858 settings for BTNS

We asked OV vendor to review the BTNS ov8858 register
settings and changing the settings according to the
feedback from OV vendor.

Change-Id: Id859b085ba128e3b29ec60da7edde47ea3832e01
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-13009
Original-Tracked-On: #H1504112909
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/88478
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Build: Dessert M COHO (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Validation-Android: Dessert M COHO (SC)
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/450389
---
 .../camera/drivers/media/i2c/ov8858_btns.h         | 627 +++++++++++----------
 1 file changed, 328 insertions(+), 299 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858_btns.h b/drivers/external_drivers/camera/drivers/media/i2c/ov8858_btns.h
index c0cd476..9233d8b 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858_btns.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858_btns.h
@@ -348,6 +348,8 @@ static struct ov8858_vcm ov8858_vcms[] = {
 
 static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x0103, 0x01}, /* software_reset */
+	{OV8858_8BIT, 0x303f, 0x01},
+	{OV8858_8BIT, 0x3012, 0x6c},
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
 	/* PLL settings */
 	{OV8858_8BIT, 0x0300, 0x05}, /* pll1_pre_div = /4 */
@@ -355,21 +357,53 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x0303, 0x00}, /* pll1_divm = /(1 + 0) */
 	{OV8858_8BIT, 0x0304, 0x03}, /* pll1_div_mipi = /8 */
 	{OV8858_8BIT, 0x030B, 0x02}, /* pll2_pre_div = /2 */
-	{OV8858_8BIT, 0x030D, 0x4E}, /* pll2_r_divp = 78 */
+	{OV8858_8BIT, 0x030D, 0x4B}, /* pll2_r_divp = 75 */
 	{OV8858_8BIT, 0x030E, 0x00}, /* pll2_r_divs = /1 */
 	{OV8858_8BIT, 0x030F, 0x04}, /* pll2_r_divsp = /(1 + 4) */
 	/* pll2_pre_div0 = /1, pll2_r_divdac = /(1 + 1) */
 	{OV8858_8BIT, 0x0312, 0x01},
 	{OV8858_8BIT, 0x031E, 0x0C}, /* pll1_no_lat = 1, mipi_bitsel_man = 0 */
 
-	/* PAD OEN2, VSYNC out enable=0x80, disable=0x00 */
-	{OV8858_8BIT, 0x3002, 0x80},
-	/* PAD OUT2, VSYNC pulse direction low-to-high = 1 */
-	{OV8858_8BIT, 0x3007, 0x01},
-	/* PAD SEL2, VSYNC out value = 0 */
-	{OV8858_8BIT, 0x300D, 0x00},
-	/* PAD OUT2, VSYNC out select = 0 */
-	{OV8858_8BIT, 0x3010, 0x00},
+	{OV8858_8BIT, 0x3600, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x3601, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x3602, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x3603, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x3604, 0x22}, /* Unknown */
+	{OV8858_8BIT, 0x3605, 0x28}, /* Unknown */
+	{OV8858_8BIT, 0x3606, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x3607, 0x20}, /* Unknown */
+	{OV8858_8BIT, 0x3608, 0x11}, /* Unknown */
+	{OV8858_8BIT, 0x3609, 0x28}, /* Unknown */
+	{OV8858_8BIT, 0x360A, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x360B, 0x05}, /* Unknown */
+	{OV8858_8BIT, 0x360C, 0xD4}, /* Unknown */
+	{OV8858_8BIT, 0x360D, 0x40}, /* Unknown */
+	{OV8858_8BIT, 0x360E, 0x0C}, /* Unknown */
+	{OV8858_8BIT, 0x360F, 0x20}, /* Unknown */
+	{OV8858_8BIT, 0x3610, 0x07}, /* Unknown */
+	{OV8858_8BIT, 0x3611, 0x20}, /* Unknown */
+	{OV8858_8BIT, 0x3612, 0x88}, /* Unknown */
+	{OV8858_8BIT, 0x3613, 0x80}, /* Unknown */
+	{OV8858_8BIT, 0x3614, 0x58}, /* Unknown */
+	{OV8858_8BIT, 0x3615, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x3616, 0x4A}, /* Unknown */
+	{OV8858_8BIT, 0x3617, 0x40}, /* Unknown */
+	{OV8858_8BIT, 0x3618, 0x5A}, /* Unknown */
+	{OV8858_8BIT, 0x3619, 0x70}, /* Unknown */
+	{OV8858_8BIT, 0x361A, 0x99}, /* Unknown */
+	{OV8858_8BIT, 0x361B, 0x0a}, /* Unknown */
+	{OV8858_8BIT, 0x361C, 0x07}, /* Unknown */
+	{OV8858_8BIT, 0x361D, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x361E, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x361F, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x3638, 0xFF}, /* Unknown */
+	{OV8858_8BIT, 0x3633, 0x0F}, /* Unknown */
+	{OV8858_8BIT, 0x3634, 0x0F}, /* Unknown */
+	{OV8858_8BIT, 0x3635, 0x0F}, /* Unknown */
+	{OV8858_8BIT, 0x3636, 0x12}, /* Unknown */
+	{OV8858_8BIT, 0x3645, 0x13}, /* Unknown */
+	{OV8858_8BIT, 0x3646, 0x83}, /* Unknown */
+	{OV8858_8BIT, 0x364A, 0x07}, /* Unknown */
 
 	/* Npump clock div = /2, Ppump clock div = /4 */
 	{OV8858_8BIT, 0x3015, 0x01},
@@ -385,13 +419,13 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	 * pd_mipi enable when rst_sync = 1
 	 */
 	{OV8858_8BIT, 0x3022, 0x01},
+
 	{OV8858_8BIT, 0x3031, 0x0A}, /* mipi_bit_sel = 10 */
 	{OV8858_8BIT, 0x3034, 0x00}, /* Unknown */
 	/* sclk_div = /1, sclk_pre_div = /1, chip debug = 1 */
 	{OV8858_8BIT, 0x3106, 0x01},
 
 	{OV8858_8BIT, 0x3305, 0xF1}, /* Unknown */
-	{OV8858_8BIT, 0x3307, 0x04}, /* Unknown */
 	{OV8858_8BIT, 0x3308, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x3309, 0x28}, /* Unknown */
 	{OV8858_8BIT, 0x330A, 0x00}, /* Unknown */
@@ -400,10 +434,11 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x330D, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x330E, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x330F, 0x40}, /* Unknown */
+	{OV8858_8BIT, 0x3307, 0x04}, /* Unknown */
 
-	{OV8858_8BIT, 0x3500, 0x00}, /* long exposure = 0x9A20 */
-	{OV8858_8BIT, 0x3501, 0x9A}, /* long exposure = 0x9A20 */
-	{OV8858_8BIT, 0x3502, 0x20}, /* long exposure = 0x9A20 */
+	{OV8858_8BIT, 0x3500, 0x00},
+	{OV8858_8BIT, 0x3501, 0x4A},
+	{OV8858_8BIT, 0x3502, 0x60},
 	/*
 	 * Digital fraction gain delay option = Delay 1 frame,
 	 * Gain change delay option = Delay 1 frame,
@@ -412,8 +447,9 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	 * Exposure delay option (must be 0 = Delay 1 frame,
 	 * Exposure change delay option (must be 0) = Delay 1 frame
 	 */
-	{OV8858_8BIT, 0x3503, 0x00},
+	{OV8858_8BIT, 0x3503, 0x80},
 	{OV8858_8BIT, 0x3505, 0x80}, /* gain conversation option */
+
 	/*
 	 * [10:7] are integer gain, [6:0] are fraction gain. For example:
 	 * 0x80 is 1x gain, 0x100 is 2x gain, 0x1C0 is 3.5x gain
@@ -426,59 +462,18 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x3511, 0x02}, /* short exposure = 0x000200 */
 	{OV8858_8BIT, 0x3512, 0x00}, /* short exposure = 0x000200 */
 
-	{OV8858_8BIT, 0x3600, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3601, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3602, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3603, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3604, 0x22}, /* Unknown */
-	{OV8858_8BIT, 0x3605, 0x30}, /* Unknown */
-	{OV8858_8BIT, 0x3606, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3607, 0x20}, /* Unknown */
-	{OV8858_8BIT, 0x3608, 0x11}, /* Unknown */
-	{OV8858_8BIT, 0x3609, 0x28}, /* Unknown */
-	{OV8858_8BIT, 0x360A, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x360B, 0x06}, /* Unknown */
-	{OV8858_8BIT, 0x360C, 0xDC}, /* Unknown */
-	{OV8858_8BIT, 0x360D, 0x40}, /* Unknown */
-	{OV8858_8BIT, 0x360E, 0x0C}, /* Unknown */
-	{OV8858_8BIT, 0x360F, 0x20}, /* Unknown */
-	{OV8858_8BIT, 0x3610, 0x07}, /* Unknown */
-	{OV8858_8BIT, 0x3611, 0x20}, /* Unknown */
-	{OV8858_8BIT, 0x3612, 0x88}, /* Unknown */
-	{OV8858_8BIT, 0x3613, 0x80}, /* Unknown */
-	{OV8858_8BIT, 0x3614, 0x58}, /* Unknown */
-	{OV8858_8BIT, 0x3615, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3616, 0x4A}, /* Unknown */
-	{OV8858_8BIT, 0x3617, 0x90}, /* Unknown */
-	{OV8858_8BIT, 0x3618, 0x56}, /* Unknown */
-	{OV8858_8BIT, 0x3619, 0x70}, /* Unknown */
-	{OV8858_8BIT, 0x361A, 0x99}, /* Unknown */
-	{OV8858_8BIT, 0x361B, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x361C, 0x07}, /* Unknown */
-	{OV8858_8BIT, 0x361D, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x361E, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x361F, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3633, 0x0C}, /* Unknown */
-	{OV8858_8BIT, 0x3634, 0x0C}, /* Unknown */
-	{OV8858_8BIT, 0x3635, 0x0C}, /* Unknown */
-	{OV8858_8BIT, 0x3636, 0x0C}, /* Unknown */
-	{OV8858_8BIT, 0x3638, 0xFF}, /* Unknown */
-	{OV8858_8BIT, 0x3645, 0x13}, /* Unknown */
-	{OV8858_8BIT, 0x3646, 0x83}, /* Unknown */
-	{OV8858_8BIT, 0x364A, 0x07}, /* Unknown */
-
 	{OV8858_8BIT, 0x3700, 0x30}, /* Unknown */
 	{OV8858_8BIT, 0x3701, 0x18}, /* Unknown */
 	{OV8858_8BIT, 0x3702, 0x50}, /* Unknown */
-	{OV8858_8BIT, 0x3703, 0x32}, /* Unknown */
+	{OV8858_8BIT, 0x3703, 0x33}, /* Unknown */
 	{OV8858_8BIT, 0x3704, 0x28}, /* Unknown */
 	{OV8858_8BIT, 0x3705, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3706, 0x6A}, /* Unknown */
+	{OV8858_8BIT, 0x3706, 0x82}, /* Unknown */
 	{OV8858_8BIT, 0x3707, 0x08}, /* Unknown */
 	{OV8858_8BIT, 0x3708, 0x48}, /* Unknown */
 	{OV8858_8BIT, 0x3709, 0x66}, /* Unknown */
 	{OV8858_8BIT, 0x370A, 0x01}, /* Unknown */
-	{OV8858_8BIT, 0x370B, 0x6A}, /* Unknown */
+	{OV8858_8BIT, 0x370B, 0x82}, /* Unknown */
 	{OV8858_8BIT, 0x370C, 0x07}, /* Unknown */
 	{OV8858_8BIT, 0x3712, 0x44}, /* Unknown */
 	{OV8858_8BIT, 0x3714, 0x24}, /* Unknown */
@@ -509,6 +504,8 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x373B, 0x0B}, /* Unknown */
 	{OV8858_8BIT, 0x373C, 0x14}, /* Unknown */
 	{OV8858_8BIT, 0x373E, 0x06}, /* Unknown */
+	{OV8858_8BIT, 0x3750, 0x0A}, /* Unknown */
+	{OV8858_8BIT, 0x3751, 0x0E}, /* Unknown */
 	{OV8858_8BIT, 0x3755, 0x10}, /* Unknown */
 	{OV8858_8BIT, 0x3758, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x3759, 0x4C}, /* Unknown */
@@ -521,11 +518,11 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x3760, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x3761, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x3762, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3763, 0x00}, /* Unknown */
+	{OV8858_8BIT, 0x3763, 0x18}, /* Unknown */
 	{OV8858_8BIT, 0x3766, 0xFF}, /* Unknown */
 	{OV8858_8BIT, 0x3768, 0x22}, /* Unknown */
-	{OV8858_8BIT, 0x3769, 0x44}, /* Unknown */
-	{OV8858_8BIT, 0x376A, 0x44}, /* Unknown */
+	{OV8858_8BIT, 0x3769, 0x66}, /* Unknown */
+	{OV8858_8BIT, 0x376A, 0x24}, /* Unknown */
 	{OV8858_8BIT, 0x376B, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x376F, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x3772, 0x46}, /* Unknown */
@@ -534,7 +531,8 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x3775, 0x13}, /* Unknown */
 	{OV8858_8BIT, 0x3776, 0x08}, /* Unknown */
 	{OV8858_8BIT, 0x3777, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x3778, 0x16}, /* Unknown */
+	{OV8858_8BIT, 0x3778, 0x17}, /* Unknown */
+	{OV8858_8BIT, 0x377A, 0x0C}, /* Unknown */
 	{OV8858_8BIT, 0x37A0, 0x88}, /* Unknown */
 	{OV8858_8BIT, 0x37A1, 0x7A}, /* Unknown */
 	{OV8858_8BIT, 0x37A2, 0x7A}, /* Unknown */
@@ -556,7 +554,7 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x37B2, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x37B3, 0x84}, /* Unknown */
 	{OV8858_8BIT, 0x37B4, 0x84}, /* Unknown */
-	{OV8858_8BIT, 0x37B5, 0x66}, /* Unknown */
+	{OV8858_8BIT, 0x37B5, 0x60}, /* Unknown */
 	{OV8858_8BIT, 0x37B6, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x37B7, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x37B8, 0x00}, /* Unknown */
@@ -575,36 +573,35 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x380A, 0x09}, /* v_output_size high */
 	{OV8858_8BIT, 0x380B, 0x90}, /* v_output_size low */
 	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x0A}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0x0D}, /* vertical timing size low */
+	{OV8858_8BIT, 0x380D, 0x90}, /* horizontal timing size low */
+	{OV8858_8BIT, 0x380E, 0x09}, /* vertical timing size high */
+	{OV8858_8BIT, 0x380F, 0xAE}, /* vertical timing size low */
 	{OV8858_8BIT, 0x3810, 0x00}, /* h_win offset high */
 	{OV8858_8BIT, 0x3811, 0x04}, /* h_win offset low */
-	{OV8858_8BIT, 0x3812, 0x00}, /* v_win offset high */
-	{OV8858_8BIT, 0x3813, 0x02}, /* v_win offset low */
+	{OV8858_8BIT, 0x3812, 0x02}, /* v_win offset high */
+	{OV8858_8BIT, 0x3813, 0x01}, /* v_win offset low */
 	{OV8858_8BIT, 0x3814, 0x01}, /* h_odd_inc */
 	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
 	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
-	{OV8858_8BIT, 0x3821, 0x40}, /* format2 */
+	{OV8858_8BIT, 0x3821, 0x46}, /* format2 */
 	{OV8858_8BIT, 0x382A, 0x01}, /* v_odd_inc */
 	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-
+	{OV8858_8BIT, 0x382D, 0xFF},
 	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
+	{OV8858_8BIT, 0x3836, 0x02}, /* Unknown */
 	{OV8858_8BIT, 0x3837, 0x18}, /* Unknown */
 	{OV8858_8BIT, 0x3841, 0xFF}, /* AUTO_SIZE_CTRL */
 	{OV8858_8BIT, 0x3846, 0x48}, /* Unknown */
 
-	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
+	{OV8858_8BIT, 0x3D85, 0x16}, /* OTP_REG85 */
 	{OV8858_8BIT, 0x3D8C, 0x73}, /* OTP_SETTING_STT_ADDRESS */
 	{OV8858_8BIT, 0x3D8D, 0xDE}, /* OTP_SETTING_STT_ADDRESS */
 	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
-	{OV8858_8BIT, 0x3F0A, 0x80}, /* PSRAM control register */
 
 	{OV8858_8BIT, 0x4000, 0xF1}, /* BLC CTRL00 = default */
 	{OV8858_8BIT, 0x4001, 0x00}, /* BLC CTRL01 */
-	{OV8858_8BIT, 0x4002, 0x27}, /* BLC offset = 0x27 */
 	{OV8858_8BIT, 0x4005, 0x10}, /* BLC target = 0x0010 */
+	{OV8858_8BIT, 0x4002, 0x27}, /* BLC offset = 0x27 */
 	{OV8858_8BIT, 0x4009, 0x81}, /* BLC CTRL09 */
 	{OV8858_8BIT, 0x400B, 0x0C}, /* BLC CTRL0B = default */
 	{OV8858_8BIT, 0x400A, 0x01},
@@ -634,19 +631,18 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x4300, 0xFF}, /* clip_max[11:4] = 0xFFF */
 	{OV8858_8BIT, 0x4301, 0x00}, /* clip_min[11:4] = 0 */
 	{OV8858_8BIT, 0x4302, 0x0F}, /* clip_min/max[3:0] */
-	{OV8858_8BIT, 0x4307, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x4316, 0x00}, /* CTRL16 = default */
 	{OV8858_8BIT, 0x4503, 0x18}, /* Unknown */
-	{OV8858_8BIT, 0x4500, 0x38}, /* Unknown */
+/*	{OV8858_8BIT, 0x4500, 0x38}, */ /* Unknown */
 	{OV8858_8BIT, 0x4600, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
 	/* wkup_dly = Mark1 wakeup delay/2^10 = 0x25 */
 	{OV8858_8BIT, 0x4808, 0x25},
 	{OV8858_8BIT, 0x4816, 0x52}, /* Embedded data type*/
 	{OV8858_8BIT, 0x481F, 0x32}, /* clk_prepare_min = 0x32 */
+	{OV8858_8BIT, 0x4837, 0x13}, /* pclk_period = 0x13 */
 	{OV8858_8BIT, 0x4825, 0x3A}, /* lpx_p_min = 0x3A */
 	{OV8858_8BIT, 0x4826, 0x40}, /* hs_prepare_min = 0x40 */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
 	{OV8858_8BIT, 0x4850, 0x10}, /* LANE SEL01 */
 	{OV8858_8BIT, 0x4851, 0x32}, /* LANE SEL02 */
 
@@ -701,16 +697,33 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	 * Black/white sensor mode enable = 0
 	 * Manual mode enable = 0
 	 */
-	{OV8858_8BIT, 0x5780, 0xFC},
-	{OV8858_8BIT, 0x5784, 0x0C}, /* DPC CTRL04 */
-	{OV8858_8BIT, 0x5787, 0x40}, /* DPC CTRL07 */
-	{OV8858_8BIT, 0x5788, 0x08}, /* DPC CTRL08 */
-	{OV8858_8BIT, 0x578A, 0x02}, /* DPC CTRL0A */
-	{OV8858_8BIT, 0x578B, 0x01}, /* DPC CTRL0B */
-	{OV8858_8BIT, 0x578C, 0x01}, /* DPC CTRL0C */
-	{OV8858_8BIT, 0x578E, 0x02}, /* DPC CTRL0E */
-	{OV8858_8BIT, 0x578F, 0x01}, /* DPC CTRL0F */
-	{OV8858_8BIT, 0x5790, 0x01}, /* DPC CTRL10 */
+	{OV8858_8BIT, 0x5780, 0x3E},
+	{OV8858_8BIT, 0x5781, 0x0F},
+	{OV8858_8BIT, 0x5782, 0x44},
+	{OV8858_8BIT, 0x5783, 0x02},
+	{OV8858_8BIT, 0x5784, 0x01}, /* DPC CTRL04 */
+	{OV8858_8BIT, 0x5785, 0x00}, /* DPC CTRL05 */
+	{OV8858_8BIT, 0x5786, 0x00}, /* DPC CTRL06 */
+	{OV8858_8BIT, 0x5787, 0x04}, /* DPC CTRL07 */
+	{OV8858_8BIT, 0x5788, 0x02}, /* DPC CTRL08 */
+	{OV8858_8BIT, 0x5789, 0x0F}, /* DPC CTRL09 */
+	{OV8858_8BIT, 0x578A, 0xFD}, /* DPC CTRL0A */
+	{OV8858_8BIT, 0x578B, 0xF5}, /* DPC CTRL0B */
+	{OV8858_8BIT, 0x578C, 0xF5}, /* DPC CTRL0C */
+	{OV8858_8BIT, 0x578D, 0x03}, /* DPC CTRL0D */
+	{OV8858_8BIT, 0x578E, 0x08}, /* DPC CTRL0E */
+	{OV8858_8BIT, 0x578F, 0x0C}, /* DPC CTRL0F */
+	{OV8858_8BIT, 0x5790, 0x08}, /* DPC CTRL10 */
+	{OV8858_8BIT, 0x5791, 0x04}, /* DPC CTRL11 */
+	{OV8858_8BIT, 0x5792, 0x00}, /* DPC CTRL12 */
+	{OV8858_8BIT, 0x5793, 0x52}, /* DPC CTRL13 */
+	{OV8858_8BIT, 0x5794, 0xA3}, /* DPC CTRL14 */
+	{OV8858_8BIT, 0x5871, 0x0D},
+	{OV8858_8BIT, 0x5870, 0x18},
+	{OV8858_8BIT, 0x586E, 0x10},
+	{OV8858_8BIT, 0x586F, 0x08},
+	{OV8858_8BIT, 0x58F8, 0x3D},
+
 	{OV8858_8BIT, 0x5901, 0x00}, /* VAP CTRL01 = default */
 	/* WINC CTRL08 = embedded data in 1st line*/
 	{OV8858_8BIT, 0x5A08, 0x00},
@@ -722,6 +735,14 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x5E00, 0x00}, /* PRE CTRL00 = default */
 	{OV8858_8BIT, 0x5E01, 0x41}, /* PRE_CTRL01 = default */
 
+	{OV8858_8BIT, 0x470B, 0x28},
+	{OV8858_8BIT, 0x4202, 0x00},
+	{OV8858_8BIT, 0x400D, 0x10},
+	{OV8858_8BIT, 0x4040, 0x07},
+	{OV8858_8BIT, 0x403E, 0x08},
+	{OV8858_8BIT, 0x4041, 0xC6},
+	{OV8858_8BIT, 0x3007, 0x80},
+
 	{OV8858_TOK_TERM, 0, 0}
 };
 
@@ -729,155 +750,109 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 
 static const struct ov8858_reg ov8858_8M[] = {
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
-	{OV8858_8BIT, 0x3778, 0x16}, /* Unknown */
+	{OV8858_8BIT, 0x3841, 0x00},
 	{OV8858_8BIT, 0x3800, 0x00}, /* h_crop_start high */
-	{OV8858_8BIT, 0x3801, 0x0C}, /* h_crop_start low 12 */
+	{OV8858_8BIT, 0x3801, 0x00}, /* h_crop_start low */
 	{OV8858_8BIT, 0x3802, 0x00}, /* v_crop_start high */
-	{OV8858_8BIT, 0x3803, 0x0C}, /* v_crop_start low */
+	{OV8858_8BIT, 0x3803, 0x00}, /* v_crop_start low */
 	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high */
-	{OV8858_8BIT, 0x3805, 0xD3}, /* h_crop_end low 3283 */
+	{OV8858_8BIT, 0x3805, 0xDF}, /* h_crop_end low */
 	{OV8858_8BIT, 0x3806, 0x09}, /* v_crop_end high */
-	{OV8858_8BIT, 0x3807, 0xA3}, /* v_crop_end low */
+	{OV8858_8BIT, 0x3807, 0xAF}, /* v_crop_end low */
 	{OV8858_8BIT, 0x3808, 0x0C}, /* h_output_size high 3280 x 2464 */
 	{OV8858_8BIT, 0x3809, 0xD0}, /* h_output_size low */
 	{OV8858_8BIT, 0x380A, 0x09}, /* v_output_size high */
 	{OV8858_8BIT, 0x380B, 0xa0}, /* v_output_size low */
 	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x0A}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0x0D}, /* vertical timing size low */
+	{OV8858_8BIT, 0x380D, 0xC0}, /* horizontal timing size low */
+	{OV8858_8BIT, 0x380E, 0x09}, /* vertical timing size high */
+	{OV8858_8BIT, 0x380F, 0xCE}, /* vertical timing size low */
+	{OV8858_8BIT, 0x3810, 0x00},
+	{OV8858_8BIT, 0x3811, 0x07},
+	{OV8858_8BIT, 0x3812, 0x00},
+	{OV8858_8BIT, 0x3813, 0x08},
 	{OV8858_8BIT, 0x3814, 0x01}, /* h_odd_inc */
 	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
-	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
-	{OV8858_8BIT, 0x3821, 0x40}, /* format2 */
+	{OV8858_8BIT, 0x3821, 0x46}, /* format2 */
 	{OV8858_8BIT, 0x382A, 0x01}, /* v_odd_inc */
 	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
-	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
-	{OV8858_8BIT, 0x4034, 0x3F}, /* Unknown */
-	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
+	{OV8858_8BIT, 0x382D, 0xFF},
+
+	{OV8858_8BIT, 0x4001, 0x00},
+	{OV8858_8BIT, 0x4022, 0x0C},
+	{OV8858_8BIT, 0x4023, 0x60},
+	{OV8858_8BIT, 0x4024, 0x0F},
+	{OV8858_8BIT, 0x4025, 0x36},
+	{OV8858_8BIT, 0x4026, 0x0F},
+	{OV8858_8BIT, 0x4027, 0x37},
+	{OV8858_8BIT, 0x4028, 0x00},
+	{OV8858_8BIT, 0x4029, 0x04},
+	{OV8858_8BIT, 0x402a, 0x04},
+	{OV8858_8BIT, 0x402b, 0x08},
+	{OV8858_8BIT, 0x402c, 0x00},
+	{OV8858_8BIT, 0x402d, 0x02},
+	{OV8858_8BIT, 0x402e, 0x04},
+	{OV8858_8BIT, 0x402f, 0x08},
 	{OV8858_8BIT, 0x4600, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
-	{OV8858_TOK_TERM, 0, 0}
-};
 
-static const struct ov8858_reg ov8858_3276x1848[] = {
-	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
-	{OV8858_8BIT, 0x3778, 0x16}, /* Unknown */
-	{OV8858_8BIT, 0x3800, 0x00}, /* h_crop_start high */
-	{OV8858_8BIT, 0x3801, 0x10}, /* h_crop_start low  0c->10*/
-	{OV8858_8BIT, 0x3802, 0x01}, /* v_crop_start high */
-	{OV8858_8BIT, 0x3803, 0x42}, /* v_crop_start low 3e->42*/
-	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high */
-	{OV8858_8BIT, 0x3805, 0xD3}, /* h_crop_end low */
-	{OV8858_8BIT, 0x3806, 0x08}, /* v_crop_end high */
-	{OV8858_8BIT, 0x3807, 0x71}, /* v_crop_end low */
-	{OV8858_8BIT, 0x3808, 0x0C}, /* h_output_size high 3276 x 1848 */
-	{OV8858_8BIT, 0x3809, 0xCC}, /* h_output_size low d0->cc*/
-	{OV8858_8BIT, 0x380A, 0x07}, /* v_output_size high */
-	{OV8858_8BIT, 0x380B, 0x38}, /* v_output_size low 3c->38*/
-	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x0A}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0x0D}, /* vertical timing size low */
-	{OV8858_8BIT, 0x3814, 0x01}, /* h_odd_inc */
-	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
-	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
-	{OV8858_8BIT, 0x3821, 0x40}, /* format2 */
-	{OV8858_8BIT, 0x382A, 0x01}, /* v_odd_inc */
-	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
-	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
-	{OV8858_8BIT, 0x4034, 0x3F}, /* Unknown */
-	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
-	{OV8858_8BIT, 0x4600, 0x01}, /* Unknown */
-	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
 static const struct ov8858_reg ov8858_6M[] = {
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
-	{OV8858_8BIT, 0x3778, 0x16}, /* Unknown */
+	{OV8858_8BIT, 0x3841, 0x00},
 	{OV8858_8BIT, 0x3800, 0x00}, /* h_crop_start high */
-	{OV8858_8BIT, 0x3801, 0x0C}, /* h_crop_start low */
+	{OV8858_8BIT, 0x3801, 0x00}, /* h_crop_start low */
 	{OV8858_8BIT, 0x3802, 0x01}, /* v_crop_start high */
-	{OV8858_8BIT, 0x3803, 0x3E}, /* v_crop_start low */
+	{OV8858_8BIT, 0x3803, 0x32}, /* v_crop_start low */
 	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high */
-	{OV8858_8BIT, 0x3805, 0xD3}, /* h_crop_end low */
+	{OV8858_8BIT, 0x3805, 0xDF}, /* h_crop_end low */
 	{OV8858_8BIT, 0x3806, 0x08}, /* v_crop_end high */
-	{OV8858_8BIT, 0x3807, 0x71}, /* v_crop_end low */
+	{OV8858_8BIT, 0x3807, 0x7D}, /* v_crop_end low */
 	{OV8858_8BIT, 0x3808, 0x0C}, /* h_output_size high 3280 x 1852 */
 	{OV8858_8BIT, 0x3809, 0xD0}, /* h_output_size low */
 	{OV8858_8BIT, 0x380A, 0x07}, /* v_output_size high */
 	{OV8858_8BIT, 0x380B, 0x3C}, /* v_output_size low */
 	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x0A}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0x0D}, /* vertical timing size low */
+	{OV8858_8BIT, 0x380D, 0xA0}, /* horizontal timing size low */
+	{OV8858_8BIT, 0x380E, 0x09}, /* vertical timing size high */
+	{OV8858_8BIT, 0x380F, 0xAE}, /* vertical timing size low */
+	{OV8858_8BIT, 0x3810, 0x00},
+	{OV8858_8BIT, 0x3811, 0x09},
+	{OV8858_8BIT, 0x3812, 0x00},
+	{OV8858_8BIT, 0x3813, 0x08},
 	{OV8858_8BIT, 0x3814, 0x01}, /* h_odd_inc */
 	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
-	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
-	{OV8858_8BIT, 0x3821, 0x40}, /* format2 */
+	{OV8858_8BIT, 0x3821, 0x46}, /* format2 */
 	{OV8858_8BIT, 0x382A, 0x01}, /* v_odd_inc */
 	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
-	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
-	{OV8858_8BIT, 0x4034, 0x3F}, /* Unknown */
-	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
+	{OV8858_8BIT, 0x382D, 0xFF},
+
+	{OV8858_8BIT, 0x4001, 0x00},
+	{OV8858_8BIT, 0x4022, 0x0C},
+	{OV8858_8BIT, 0x4023, 0x60},
+	{OV8858_8BIT, 0x4024, 0x0F},
+	{OV8858_8BIT, 0x4025, 0x36},
+	{OV8858_8BIT, 0x4026, 0x0F},
+	{OV8858_8BIT, 0x4027, 0x37},
+	{OV8858_8BIT, 0x4028, 0x00},
+	{OV8858_8BIT, 0x4029, 0x04},
+	{OV8858_8BIT, 0x402a, 0x04},
+	{OV8858_8BIT, 0x402b, 0x08},
+	{OV8858_8BIT, 0x402c, 0x00},
+	{OV8858_8BIT, 0x402d, 0x02},
+	{OV8858_8BIT, 0x402e, 0x04},
+	{OV8858_8BIT, 0x402f, 0x08},
 	{OV8858_8BIT, 0x4600, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
-	{OV8858_TOK_TERM, 0, 0}
-};
 
-static const struct ov8858_reg ov8858_1080P_60[] = {
-	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
-	{OV8858_8BIT, 0x3778, 0x17}, /* Unknown */
-	{OV8858_8BIT, 0x3800, 0x02}, /* h_crop_start high */
-	{OV8858_8BIT, 0x3801, 0x26}, /* h_crop_start low */
-	{OV8858_8BIT, 0x3802, 0x02}, /* v_crop_start high */
-	{OV8858_8BIT, 0x3803, 0x8C}, /* v_crop_start low */
-	{OV8858_8BIT, 0x3804, 0x0A}, /* h_crop_end high */
-	{OV8858_8BIT, 0x3805, 0x9D}, /* h_crop_end low */
-	{OV8858_8BIT, 0x3806, 0x07}, /* v_crop_end high */
-	{OV8858_8BIT, 0x3807, 0x0A}, /* v_crop_end low */
-	{OV8858_8BIT, 0x3808, 0x07}, /* h_output_size high*/
-	{OV8858_8BIT, 0x3809, 0x90}, /* h_output_size low */
-	{OV8858_8BIT, 0x380A, 0x04}, /* v_output_size high */
-	{OV8858_8BIT, 0x380B, 0x48}, /* v_output_size low */
-	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x04}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0xEC}, /* vertical timing size low */
-	{OV8858_8BIT, 0x3814, 0x01}, /* h_odd_inc */
-	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
-	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
-	{OV8858_8BIT, 0x3821, 0x40}, /* format2 */
-	{OV8858_8BIT, 0x382A, 0x01}, /* v_odd_inc */
-	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
-	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
-	{OV8858_8BIT, 0x4034, 0x3F}, /* Unknown */
-	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
-	{OV8858_8BIT, 0x4600, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x4601, 0xef}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x16}, /* pclk_period = 0x16 */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
-static const struct ov8858_reg ov8858_1080P_30[] = {
+static const struct ov8858_reg ov8858_1080P[] = {
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
-	{OV8858_8BIT, 0x3778, 0x17}, /* Unknown */
+	{OV8858_8BIT, 0x3841, 0xFF},
 	{OV8858_8BIT, 0x3800, 0x02}, /* h_crop_start high */
 	{OV8858_8BIT, 0x3801, 0x26}, /* h_crop_start low */
 	{OV8858_8BIT, 0x3802, 0x02}, /* v_crop_start high */
@@ -891,136 +866,190 @@ static const struct ov8858_reg ov8858_1080P_30[] = {
 	{OV8858_8BIT, 0x380A, 0x04}, /* v_output_size high */
 	{OV8858_8BIT, 0x380B, 0x48}, /* v_output_size low */
 	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x0A}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0x0D}, /* vertical timing size low */
+	{OV8858_8BIT, 0x380D, 0x90}, /* horizontal timing size low */
+	{OV8858_8BIT, 0x380E, 0x09}, /* vertical timing size high */
+	{OV8858_8BIT, 0x380F, 0xAE}, /* vertical timing size low */
+	{OV8858_8BIT, 0x3810, 0x00},
+	{OV8858_8BIT, 0x3811, 0x04},
+	{OV8858_8BIT, 0x3812, 0x00},
+	{OV8858_8BIT, 0x3813, 0x02},
 	{OV8858_8BIT, 0x3814, 0x01}, /* h_odd_inc */
 	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
-	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
-	{OV8858_8BIT, 0x3821, 0x40}, /* format2 */
+	{OV8858_8BIT, 0x3821, 0x46}, /* format2 */
 	{OV8858_8BIT, 0x382A, 0x01}, /* v_odd_inc */
 	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
-	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
-	{OV8858_8BIT, 0x4034, 0x3F}, /* Unknown */
-	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
+	{OV8858_8BIT, 0x382D, 0xFF},
+
+	{OV8858_8BIT, 0x4001, 0x00},
+	{OV8858_8BIT, 0x4022, 0x07},
+	{OV8858_8BIT, 0x4023, 0x20},
+	{OV8858_8BIT, 0x4024, 0x0F},
+	{OV8858_8BIT, 0x4025, 0x36},
+	{OV8858_8BIT, 0x4026, 0x0F},
+	{OV8858_8BIT, 0x4027, 0x37},
+	{OV8858_8BIT, 0x4028, 0x00},
+	{OV8858_8BIT, 0x4029, 0x04},
+	{OV8858_8BIT, 0x402a, 0x04},
+	{OV8858_8BIT, 0x402b, 0x08},
+	{OV8858_8BIT, 0x402c, 0x00},
+	{OV8858_8BIT, 0x402d, 0x02},
+	{OV8858_8BIT, 0x402e, 0x04},
+	{OV8858_8BIT, 0x402f, 0x08},
 	{OV8858_8BIT, 0x4600, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x4601, 0xef}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x16}, /* pclk_period = 0x16 */
+	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
+
 	{OV8858_TOK_TERM, 0, 0}
 };
 
 static const struct ov8858_reg ov8858_1640x1232[] = {
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
-	{OV8858_8BIT, 0x3778, 0x16}, /* Unknown */
+	{OV8858_8BIT, 0x3841, 0x00},
 	{OV8858_8BIT, 0x3800, 0x00}, /* h_crop_start high */
-	{OV8858_8BIT, 0x3801, 0x0C}, /* h_crop_start low 12 */
+	{OV8858_8BIT, 0x3801, 0x00}, /* h_crop_start low */
 	{OV8858_8BIT, 0x3802, 0x00}, /* v_crop_start high */
-	{OV8858_8BIT, 0x3803, 0x0C}, /* v_crop_start low */
-	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high 3283 */
-	{OV8858_8BIT, 0x3805, 0xD3}, /* h_crop_end low */
+	{OV8858_8BIT, 0x3803, 0x00}, /* v_crop_start low */
+	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high */
+	{OV8858_8BIT, 0x3805, 0xDF}, /* h_crop_end low */
 	{OV8858_8BIT, 0x3806, 0x09}, /* v_crop_end high */
-	{OV8858_8BIT, 0x3807, 0xA3}, /* v_crop_end low */
+	{OV8858_8BIT, 0x3807, 0xAF}, /* v_crop_end low */
 	{OV8858_8BIT, 0x3808, 0x06}, /* h_output_size high 1640 x 1232 */
 	{OV8858_8BIT, 0x3809, 0x68}, /* h_output_size low */
 	{OV8858_8BIT, 0x380A, 0x04}, /* v_output_size high */
 	{OV8858_8BIT, 0x380B, 0xD0}, /* v_output_size low */
-	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x09}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0xAA}, /* vertical timing size low */
+	{OV8858_8BIT, 0x380C, 0x0E}, /* horizontal timing size high */
+	{OV8858_8BIT, 0x380D, 0xE0}, /* horizontal timing size low */
+	{OV8858_8BIT, 0x380E, 0x04}, /* vertical timing size high */
+	{OV8858_8BIT, 0x380F, 0xEC}, /* vertical timing size low */
+	{OV8858_8BIT, 0x3810, 0x00},
+	{OV8858_8BIT, 0x3811, 0x03},
+	{OV8858_8BIT, 0x3812, 0x00},
+	{OV8858_8BIT, 0x3813, 0x04},
 	{OV8858_8BIT, 0x3814, 0x03}, /* h_odd_inc */
 	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
-	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
 	{OV8858_8BIT, 0x3821, 0x67}, /* format2 */
 	{OV8858_8BIT, 0x382A, 0x03}, /* v_odd_inc */
 	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-	{OV8858_8BIT, 0x3830, 0x08}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x02}, /* Unknown */
-	{OV8858_8BIT, 0x3D85, 0x16}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x08}, /* PSRAM control register */
-	{OV8858_8BIT, 0x4034, 0x3F}, /* Unknown */
-	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
+	{OV8858_8BIT, 0x382D, 0x7F},
+
+	{OV8858_8BIT, 0x4001, 0x10},
+	{OV8858_8BIT, 0x4022, 0x06},
+	{OV8858_8BIT, 0x4023, 0x00},
+	{OV8858_8BIT, 0x4024, 0x0F},
+	{OV8858_8BIT, 0x4025, 0x2A},
+	{OV8858_8BIT, 0x4026, 0x0F},
+	{OV8858_8BIT, 0x4027, 0x2B},
+	{OV8858_8BIT, 0x4028, 0x00},
+	{OV8858_8BIT, 0x4029, 0x02},
+	{OV8858_8BIT, 0x402a, 0x02},
+	{OV8858_8BIT, 0x402b, 0x04},
+	{OV8858_8BIT, 0x402c, 0x00},
+	{OV8858_8BIT, 0x402d, 0x02},
+	{OV8858_8BIT, 0x402e, 0x02},
+	{OV8858_8BIT, 0x402f, 0x04},
 	{OV8858_8BIT, 0x4600, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x4601, 0xCB}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
 static const struct ov8858_reg ov8858_1640x1096[] = {
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
-	{OV8858_8BIT, 0x3778, 0x16}, /* Unknown */
+	{OV8858_8BIT, 0x3841, 0x00},
 	{OV8858_8BIT, 0x3800, 0x00}, /* h_crop_start high */
-	{OV8858_8BIT, 0x3801, 0x0C}, /* h_crop_start low 12 */
+	{OV8858_8BIT, 0x3801, 0x00}, /* h_crop_start low */
 	{OV8858_8BIT, 0x3802, 0x00}, /* v_crop_start high */
-	{OV8858_8BIT, 0x3803, 0x0C}, /* v_crop_start low */
-	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high 3283 */
-	{OV8858_8BIT, 0x3805, 0xD3}, /* h_crop_end low */
+	{OV8858_8BIT, 0x3803, 0x00}, /* v_crop_start low */
+	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high */
+	{OV8858_8BIT, 0x3805, 0xDF}, /* h_crop_end low */
 	{OV8858_8BIT, 0x3806, 0x09}, /* v_crop_end high */
-	{OV8858_8BIT, 0x3807, 0xA3}, /* v_crop_end low */
+	{OV8858_8BIT, 0x3807, 0xAF}, /* v_crop_end low */
 	{OV8858_8BIT, 0x3808, 0x06}, /* h_output_size high 1640 x 1096 */
 	{OV8858_8BIT, 0x3809, 0x68}, /* h_output_size low */
 	{OV8858_8BIT, 0x380A, 0x04}, /* v_output_size high */
 	{OV8858_8BIT, 0x380B, 0x48}, /* v_output_size low */
-	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x09}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0xAA}, /* vertical timing size low */
+	{OV8858_8BIT, 0x380C, 0x0E}, /* horizontal timing size high */
+	{OV8858_8BIT, 0x380D, 0xE0}, /* horizontal timing size low */
+	{OV8858_8BIT, 0x380E, 0x04}, /* vertical timing size high */
+	{OV8858_8BIT, 0x380F, 0xEC}, /* vertical timing size low */
+	{OV8858_8BIT, 0x3810, 0x00},
+	{OV8858_8BIT, 0x3811, 0x03},
+	{OV8858_8BIT, 0x3812, 0x00},
+	{OV8858_8BIT, 0x3813, 0x48},
 	{OV8858_8BIT, 0x3814, 0x03}, /* h_odd_inc */
 	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
-	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
 	{OV8858_8BIT, 0x3821, 0x67}, /* format2 */
 	{OV8858_8BIT, 0x382A, 0x03}, /* v_odd_inc */
 	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-	{OV8858_8BIT, 0x3830, 0x08}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x02}, /* Unknown */
-	{OV8858_8BIT, 0x3D85, 0x16}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x08}, /* PSRAM control register */
-	{OV8858_8BIT, 0x4034, 0x3F}, /* Unknown */
-	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
+	{OV8858_8BIT, 0x382D, 0x7F},
+
+	{OV8858_8BIT, 0x4001, 0x10},
+	{OV8858_8BIT, 0x4022, 0x06},
+	{OV8858_8BIT, 0x4023, 0x00},
+	{OV8858_8BIT, 0x4024, 0x0f},
+	{OV8858_8BIT, 0x4025, 0x2A},
+	{OV8858_8BIT, 0x4026, 0x0F},
+	{OV8858_8BIT, 0x4027, 0x2B},
+	{OV8858_8BIT, 0x4028, 0x00},
+	{OV8858_8BIT, 0x4029, 0x02},
+	{OV8858_8BIT, 0x402a, 0x02},
+	{OV8858_8BIT, 0x402b, 0x04},
+	{OV8858_8BIT, 0x402c, 0x00},
+	{OV8858_8BIT, 0x402d, 0x02},
+	{OV8858_8BIT, 0x402e, 0x02},
+	{OV8858_8BIT, 0x402f, 0x04},
 	{OV8858_8BIT, 0x4600, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x4601, 0xCB}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
+
 	{OV8858_TOK_TERM, 0, 0}
 };
 
 
 static const struct ov8858_reg ov8858_1640x926[] = {
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
-	{OV8858_8BIT, 0x3778, 0x16}, /* Unknown */
+	{OV8858_8BIT, 0x3841, 0x00},
 	{OV8858_8BIT, 0x3800, 0x00}, /* h_crop_start high */
-	{OV8858_8BIT, 0x3801, 0x0C}, /* h_crop_start low */
+	{OV8858_8BIT, 0x3801, 0x00}, /* h_crop_start low */
 	{OV8858_8BIT, 0x3802, 0x00},  /* v_crop_start high */
-	{OV8858_8BIT, 0x3803, 0x0C}, /* v_crop_start low */
+	{OV8858_8BIT, 0x3803, 0x00}, /* v_crop_start low */
 	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high */
-	{OV8858_8BIT, 0x3805, 0xD3}, /* h_crop_end low */
+	{OV8858_8BIT, 0x3805, 0xDF}, /* h_crop_end low */
 	{OV8858_8BIT, 0x3806, 0x09}, /* v_crop_end high */
-	{OV8858_8BIT, 0x3807, 0xA3}, /* v_crop_end low */
+	{OV8858_8BIT, 0x3807, 0xAF}, /* v_crop_end low */
 	{OV8858_8BIT, 0x3808, 0x06}, /* h_output_size high 1640 x 926 */
 	{OV8858_8BIT, 0x3809, 0x68}, /* h_output_size low */
 	{OV8858_8BIT, 0x380A, 0x03}, /* v_output_size high */
 	{OV8858_8BIT, 0x380B, 0x9E}, /* v_output_size low */
-	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
-	{OV8858_8BIT, 0x380D, 0x94}, /* horizontal timing size low */
-	{OV8858_8BIT, 0x380E, 0x09}, /* vertical timing size high */
-	{OV8858_8BIT, 0x380F, 0xAA}, /* vertical timing size low */
+	{OV8858_8BIT, 0x380C, 0x0E}, /* horizontal timing size high */
+	{OV8858_8BIT, 0x380D, 0xE0}, /* horizontal timing size low */
+	{OV8858_8BIT, 0x380E, 0x04}, /* vertical timing size high */
+	{OV8858_8BIT, 0x380F, 0xEC}, /* vertical timing size low */
+	{OV8858_8BIT, 0x3810, 0x00},
+	{OV8858_8BIT, 0x3811, 0x03},
+	{OV8858_8BIT, 0x3812, 0x00},
+	{OV8858_8BIT, 0x3813, 0x9C},
 	{OV8858_8BIT, 0x3814, 0x03}, /* h_odd_inc */
 	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
-	{OV8858_8BIT, 0x3820, 0x00}, /* format1 */
 	{OV8858_8BIT, 0x3821, 0x67}, /* format2 */
 	{OV8858_8BIT, 0x382A, 0x03}, /* v_odd_inc */
 	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
-	{OV8858_8BIT, 0x3830, 0x08}, /* Unknown */
-	{OV8858_8BIT, 0x3836, 0x02}, /* Unknown */
-	{OV8858_8BIT, 0x3D85, 0x16}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x08}, /* PSRAM control register */
-	{OV8858_8BIT, 0x4034, 0x3F}, /* Unknown */
-	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
+	{OV8858_8BIT, 0x382D, 0x7F},
+
+	{OV8858_8BIT, 0x4001, 0x10},
+	{OV8858_8BIT, 0x4022, 0x06},
+	{OV8858_8BIT, 0x4023, 0x00},
+	{OV8858_8BIT, 0x4024, 0x0f},
+	{OV8858_8BIT, 0x4025, 0x2A},
+	{OV8858_8BIT, 0x4026, 0x0F},
+	{OV8858_8BIT, 0x4027, 0x2B},
+	{OV8858_8BIT, 0x4028, 0x00},
+	{OV8858_8BIT, 0x4029, 0x02},
+	{OV8858_8BIT, 0x402a, 0x02},
+	{OV8858_8BIT, 0x402b, 0x04},
+	{OV8858_8BIT, 0x402c, 0x00},
+	{OV8858_8BIT, 0x402d, 0x02},
+	{OV8858_8BIT, 0x402e, 0x02},
+	{OV8858_8BIT, 0x402f, 0x04},
 	{OV8858_8BIT, 0x4600, 0x00}, /* Unknown */
-	{OV8858_8BIT, 0x4601, 0xCB}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
@@ -1037,8 +1066,8 @@ static struct ov8858_resolution ov8858_res_preview[] = {
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
@@ -1056,8 +1085,8 @@ static struct ov8858_resolution ov8858_res_preview[] = {
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
@@ -1068,34 +1097,34 @@ static struct ov8858_resolution ov8858_res_preview[] = {
 		.width = 1936,
 		.height = 1096,
 		.used = 0,
-		.regs = ov8858_1080P_30,
+		.regs = ov8858_1080P,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.skip_frames = 0,
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
 		},
 	},
 	{
-		.desc = "ov8858_3276x1848_PREVIEW",
-		.width = 3276,
-		.height = 1848,
+		.desc = "ov8858_6M_PREVIEW",
+		.width = 3280,
+		.height = 1852,
 		.used = 0,
-		.regs = ov8858_3276x1848,
+		.regs = ov8858_6M,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.skip_frames = 0,
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3904,
+				.lines_per_frame = 2510,
 			},
 			{
 			}
@@ -1113,8 +1142,8 @@ static struct ov8858_resolution ov8858_res_preview[] = {
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3968,
+				.lines_per_frame = 2510,
 			},
 			{
 			}
@@ -1135,8 +1164,8 @@ static struct ov8858_resolution ov8858_res_still[] = {
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
@@ -1154,27 +1183,27 @@ static struct ov8858_resolution ov8858_res_still[] = {
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
 		},
 	},
 	{
-		.desc = "ov8858_3276X1848_STILL",
-		.width = 3276,
-		.height = 1848,
+		.desc = "ov8858_6M_STILL",
+		.width = 3280,
+		.height = 1852,
 		.used = 0,
-		.regs = ov8858_3276x1848,
+		.regs = ov8858_6M,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.skip_frames = 1,
 		.fps_options =  {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3904,
+				.lines_per_frame = 2510,
 			},
 			{
 			}
@@ -1215,8 +1244,8 @@ static struct ov8858_resolution ov8858_res_video[] = {
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
@@ -1234,8 +1263,8 @@ static struct ov8858_resolution ov8858_res_video[] = {
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
@@ -1253,27 +1282,27 @@ static struct ov8858_resolution ov8858_res_video[] = {
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
 		},
 	},
     {
-		.desc = "ov8858_1080P_30_VIDEO",
+		.desc = "ov8858_1080P_VIDEO",
 		.width = 1936,
 		.height = 1096,
 		.used = 0,
-		.regs = ov8858_1080P_30,
+		.regs = ov8858_1080P,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.skip_frames = 1,
 		.fps_options = {
 			{
 				.fps = 30,
-				.pixels_per_line = 3880,
-				.lines_per_frame = 2573,
+				.pixels_per_line = 3872,
+				.lines_per_frame = 2478,
 			},
 			{
 			}
-- 
1.9.1

